tgui/packages/tgui/interfaces/Airlock.js

```diff
@@ -1,8 +1,417 @@
+      <Window.Content>
+        <Section fitted backgroundColor="rgba(0,0,0,0)">
+          {(!canAiControl) && (
+            <Modal
+              textAlign="center"
+              fontSize="24px">
+              {hackMessage ? hackMessage : "Airlock Controls Disabled"}
+              {!!canAiHack && (
+                <Hack />
+              )}
+            </Modal>
+          )}
+          <PowerStatus />
+          <AccessAndDoorControl />
+          <Electrify />
+        </Section>
+        <Section>
+          {"Access sensor reports the net identifer is:"} <Box inline italic>{netId}</Box>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const AccessPanelOnly = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+    name,
+  } = data;
+
+  return (
+    <Window
+      width={354}
+      height={460}
+      theme="ntos"
+      title={"Airlock - " + name}>
+      <Window.Content>
+        <AccessPanel />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const PowerStatus = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    mainTimeLeft,
+    backupTimeLeft,
+    wires,
+  } = data;
+
+  return (
+    <Section title="Power Status">
+      <LabeledList>
+        <LabeledList.Item
+          label="Main"
+          color={mainTimeLeft ? "bad" : "good"}
+          buttons={(
+            <Button
+              py={0.5}
+              width={6.7}
+              align="center"
+              color="bad"
+              icon="plug"
+              disabled={!!mainTimeLeft}
+              onClick={() => act("disruptMain")}>
+              Disrupt
+            </Button>
+          )}>
+          {mainTimeLeft ? "Offline" : "Online"}
+          {" "}
+          {(!wires.main_1 || !wires.main_2)
+            && "[Wires cut!]"
+            || (mainTimeLeft > 0
+              && `[${mainTimeLeft}s]`)}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Backup"
+          color={backupTimeLeft ? "bad": "good"}
+          buttons={(
+            <Button
+              py={0.5}
+              width={6.7}
+              align="center"
+              mt={0.5}
+              color="bad"
+              icon="plug"
+              disabled={!!backupTimeLeft}
+              onClick={() => act("disruptBackup")}>
+              Disrupt
+            </Button>
+          )}>
+          {backupTimeLeft ? "Offline" : "Online"}
+          {" "}
+          {(!wires.backup_1 || !wires.backup_2)
+            && "[Wires cut!]"
+            || (backupTimeLeft > 0
+              && `[${backupTimeLeft}s]`)}
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+const AccessAndDoorControl = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    mainTimeLeft,
+    backupTimeLeft,
+    wires,
+    idScanner,
+    boltsAreUp,
+    opened,
+    welded,
+  } = data;
+
+  const isDisabled = (data.status === 2);
```

Comment:
While these may be "magic numbers" definitely have a variable that says what each of these is somewhere in here.